backends/arm/tosa_quant_utils.py

```diff
@@ -204,6 +205,68 @@ def build_rescale_from_int32(
     return
 
 
+def rescale_nodes_to_int32(
+    nodes: list[Node], tosa_graph: ts.TosaSerializer, permute_memory_to_nhwc: bool
+) -> tuple[list[TosaSerializerTensor], float]:
+    """Rescales all 'nodes' to int32, adding suitable RESCALE ops to 'tosa_graph'.
+    The scales are adjusted using the smallest scale of all 'nodes'.
+
+    Returns a list of the rescaled nodes and the scale factor used,
+    needed by rescale_node_back_to_int8.
+    """
+
+    tensors = [TosaArg(node.args[0]) for node in nodes]
+    if permute_memory_to_nhwc:
+        NHWC_Order = [0, 2, 3, 1]
+        for tensor in tensors:
+            tensor.shape = [tensor.shape[i] for i in NHWC_Order]
+
+    qargs = [get_quant_node_args(node) for node in nodes]
+
+    # Scale the int8 quantized input to a common scale in the integer
+    # domain
+    min_scale = min([qarg.scale for qarg in qargs])
+    scales = [qarg.scale / min_scale for qarg in qargs]
+
+    rescaled_nodes: list[TosaSerializerTensor] = []
+    for tensor, qarg, scale in zip(tensors, qargs, scales):
+        rescaled_nodes.append(
+            build_rescale_to_int32(
+                tosa_graph,
+                tensor,
+                qarg.zp,
+                scale,
+            )
+        )
+    return rescaled_nodes, min_scale
+
+
+def rescale_node_back_to_int8(
```

Comment:
I guess since we support only one quant dtype and quant schema it is ok to hardcode?

Reply:
See, above comment, I'd say that it's ok for now.